renderite/generator/RenderiteGenerator/Generator.cs

```diff
@@ -140,21 +154,151 @@ private void WriteEnum(Type t)
+            }
+            
+            if(this._ilVerbose)
+                this._writer.WriteLine($"\t\t// NOTE: {type.Name} has no explicitly defined method named {method.Name}, trying {type.BaseType?.Name}.{method.Name}");
+
+            written |= WritePackFunction(type.BaseType!, type.BaseType!.GetMethod(method.Name)!);
+            return written;
+        }
+
+        Queue<string> names = new();
+        foreach (Instruction instruction in methodDef.Body.Instructions)
+        {
+            // if (this._ilVerbose)
+            // {
+            //     this._writer.Write("\t\t// ");
+            //     this._writer.WriteLine(instruction.ToString());
+            // }
+
+            if (instruction.OpCode.Code is Code.Ldfld or Code.Ldflda)
+            {
+                string name = ((FieldReference)instruction.Operand).Name;
+                names.Enqueue(name);
+                if (this._ilVerbose)
+                    this._writer.WriteLine($"\t\t// {instruction.OpCode.Code} {name}");
+            }
+
+            if (instruction.OpCode.Code is Code.Call && instruction.Operand is MethodReference callRef)
+            {
+                string typeName = callRef.DeclaringType.FullName;
+
+                if (callRef.Name == "Write" && callRef.Parameters.Count == 1)
+                {
+                    string name = names.Dequeue();
+                    this._writer.WriteLine($"\t\tipc.write(@TypeOf(self.{name}), self.{name});");
+                    written = true;
+                }
+                else if (callRef.Name == "Write" &&
+                         callRef.Parameters.All(p => p.ParameterType.Name == "Boolean"))
+                {
+                    List<string> paramsList = names.Select(n => "self." + n).ToList();
+                    while (paramsList.Count < 8)
+                    {
+                        paramsList.Add("false");
+                    }
+                    this._writer.WriteLine($"\t\tipc.write{paramsList.Count}PackedBools({string.Join(", ", paramsList)});");
+                    names.Clear();
+                    written = true;
+                }
+                else if (callRef.Name == "Read" && callRef.Parameters.Count == 1)
+                {
+                    string name = names.Dequeue();
+                    this._writer.WriteLine($"\t\tself.{name} = ipc.read(@TypeOf(self.{name}));");
+                    written = true;
+                }
+                else if (callRef.Name == "Read" &&
+                         callRef.Parameters.All(p => p.ParameterType.Name == "Boolean&"))
+                {
+                    List<string> paramsList = names.Select(n => "self." + n).ToList();
+                    while (paramsList.Count < 8)
+                    {
+                        paramsList.Add("_");
+                    }
+                    this._writer.WriteLine($"\t\t{string.Join(", ", paramsList)} = ipc.read{paramsList.Count}PackedBools();");
```

Comment:
These should forward errors

```suggestion
                    string name = names.Dequeue();
                    this._writer.WriteLine($"\t\tself.{name} = try ipc.read(@TypeOf(self.{name}));");
                    written = true;
                }
                else if (callRef.Name == "Read" &&
                         callRef.Parameters.All(p => p.ParameterType.Name == "Boolean&"))
                {
                    List<string> paramsList = names.Select(n => "self." + n).ToList();
                    while (paramsList.Count < 8)
                    {
                        paramsList.Add("_");
                    }
                    this._writer.WriteLine($"\t\t{string.Join(", ", paramsList)} = try ipc.read{paramsList.Count}PackedBools();");
```